### PR TITLE
bump all dependenciess with vulnerabilities

### LINKED ISSUE #2140: multiple fixable vulnerabilities
The following trivy scan results show fixable vulnerabilities:

```
Total: 9 (UNKNOWN: 0, LOW: 0, MEDIUM: 6, HIGH: 3, CRITICAL: 0)

┌─────────────────────────────────────────────────────────────┬────────────────┬──────────┬────────┬───────────────────┬────────────────────────┬─────────────────────────────────────────────────────────────┐
│                           Library                           │ Vulnerability  │ Severity │ Status │ Installed Version │     Fixed Version      │                            Title                            │
├─────────────────────────────────────────────────────────────┼────────────────┼──────────┼────────┼───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────────┤
│ ch.qos.logback:logback-classic (logback-classic-1.2.10.jar) │ CVE-2023-6378  │ HIGH     │ fixed  │ 1.2.10            │ 1.3.12, 1.4.12, 1.2.13 │ logback: serialization vulnerability in logback receiver    │
│                                                             │                │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2023-6378                   │
├─────────────────────────────────────────────────────────────┤                │          │        │                   │                        │                                                             │
│ ch.qos.logback:logback-core (logback-core-1.2.10.jar)       │                │          │        │                   │                        │                                                             │
│                                                             │                │          │        │                   │                        │                                                             │
├─────────────────────────────────────────────────────────────┼────────────────┼──────────┤        ├───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────────┤
│ com.nimbusds:nimbus-jose-jwt (nimbus-jose-jwt-9.24.jar)     │ CVE-2023-52428 │ MEDIUM   │        │ 9.24              │ 9.37.2                 │ Denial of Service in Connect2id Nimbus JOSE+JWT             │
│                                                             │                │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2023-52428                  │
├─────────────────────────────────────────────────────────────┼────────────────┤          │        ├───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────────┤
│ io.netty:netty-codec-http (aws-msk-iam-auth.jar)            │ CVE-2024-29025 │          │        │ 4.1.100.Final     │ 4.1.108.Final          │ Netty is an asynchronous event-driven network application   │
│                                                             │                │          │        │                   │                        │ framework fo ...                                            │
│                                                             │                │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2024-29025                  │
├─────────────────────────────────────────────────────────────┤                │          │        │                   │                        │                                                             │
│ io.netty:netty-codec-http                                   │                │          │        │                   │                        │                                                             │
│ (netty-codec-http-4.1.100.Final.jar)                        │                │          │        │                   │                        │                                                             │
│                                                             │                │          │        │                   │                        │                                                             │
├─────────────────────────────────────────────────────────────┼────────────────┤          │        ├───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────────┤
│ io.vertx:vertx-core (vertx-core-4.5.0.jar)                  │ CVE-2024-1023  │          │        │ 4.5.0             │ 4.5.2                  │ io.vertx/vertx-core: memory leak due to the use of Netty    │
│                                                             │                │          │        │                   │                        │ FastThreadLocal data structures...                          │
│                                                             │                │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2024-1023                   │
├─────────────────────────────────────────────────────────────┼────────────────┤          │        ├───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────────┤
│ org.apache.zookeeper:zookeeper (zookeeper-3.8.3.jar)        │ CVE-2024-23944 │          │        │ 3.8.3             │ 3.8.4, 3.9.2           │ Information disclosure in persistent watchers handling in   │
│                                                             │                │          │        │                   │                        │ Apache ZooKe ...                                            │
│                                                             │                │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2024-23944                  │
├─────────────────────────────────────────────────────────────┼────────────────┤          │        ├───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────────┤
│ org.bitbucket.b_c:jose4j (jose4j-0.9.3.jar)                 │ CVE-2023-51775 │          │        │ 0.9.3             │ 0.9.4                  │ jose4j: denial of service (CPU consumption) via a large p2c │
│                                                             │                │          │        │                   │                        │ (aka PBES2...                                               │
│                                                             │                │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2023-51775                  │
├─────────────────────────────────────────────────────────────┼────────────────┼──────────┤        ├───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────────┤
│ org.xerial.snappy:snappy-java (snappy-java-1.1.10.1.jar)    │ CVE-2023-43642 │ HIGH     │        │ 1.1.10.1          │ 1.1.10.4               │ snappy-java: Missing upper bound check on chunk length in   │
│                                                             │                │          │        │                   │                        │ snappy-java can lead...                                     │
│                                                             │                │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2023-43642                  │
└─────────────────────────────────────────────────────────────┴────────────────┴──────────┴────────┴───────────────────┴────────────────────────┴─────────────────────────────────────────────────────────────┘
```

Ideally these are fixed.

### PR DESCRIPTION
**Pull request by slaupster**
Bump dependencies for vulnerabilities. I've tried to keep in the major/minor track, using only patch versions where possible, but `nimbus-jose-jwt` needed to have a minor bump as well to avoid vulnerabilities.

This PR resolves #2140.

### CODE REVIEW

**Review by mdykiert — APPROVED** [review]

**Comment by slaupster** [comment]
resolved conflict with vertx which was bumped in another PR

**Review by aswinayyolath — APPROVED** [review]

**Comment by aswinayyolath** [comment]
@CCisGG could you please review and merge this change?

File: gradle.properties [hunk 1]
```diff
@@ -3,7 +3,7 @@ org.gradle.parallel=false
 org.gradle.jvmargs=-Xms512m -Xmx512m
 scalaVersion=2.13.13
 kafkaVersion=3.5.1
-zookeeperVersion=3.8.3
-nettyVersion=4.1.100.Final
-jettyVersion=9.4.53.v20231009
+zookeeperVersion=3.8.4
+nettyVersion=4.1.108.Final
+jettyVersion=9.4.54.v20240208
```
Comment (akatona84) [inline gradle.properties:8]: There's a newer version for jetty, 9.4.54 has a medium one: https://nvd.nist.gov/vuln/detail/CVE-2024-8184
9.4.56.v20240826

Comment (aswinayyolath) [reply gradle.properties:8]: I think moving to `9.4.56.v20240826` should be safe

File: gradle.properties [hunk 2]
```diff
@@ -3,7 +3,7 @@ org.gradle.parallel=false
 org.gradle.jvmargs=-Xms512m -Xmx512m
 scalaVersion=2.13.13
 kafkaVersion=3.5.1
-zookeeperVersion=3.8.3
-nettyVersion=4.1.100.Final
-jettyVersion=9.4.53.v20231009
+zookeeperVersion=3.8.4
+nettyVersion=4.1.108.Final
```
[the remark this answers is not in the document]
Comment (aswinayyolath) [reply gradle.properties:7]: there is [4.1.114.Final](https://security.snyk.io/package/maven/io.netty%3Anetty-codec/4.1.114.Final) which doesn't seem to have any CVE

**Comment by slaupster** [comment]
this has been resolved - thanks!

**Pull request closed without merging**